generated/src/aws-cpp-sdk-dynamodb/include/aws/dynamodb/DynamoDBClient.h

```diff
@@ -33,10 +36,16 @@ namespace DynamoDB
    * Zones in an Amazon Web Services Region, providing built-in high availability and
    * data durability.</p>
    */
-  class AWS_DYNAMODB_API DynamoDBClient : public Aws::Client::AWSJsonClient, public Aws::Client::ClientWithAsyncTemplateMethods<DynamoDBClient>
+  class AWS_DYNAMODB_API DynamoDBClient : smithy::client::AwsSmithyClientT<Aws::DynamoDB::SERVICE_NAME,
```

Comment:
might worth adding common wrappers such as

```
    template<const char* ServiceNameT,
             typename ServiceClientConfigurationT,
             typename EndpointProviderT>
    class AwsJsonSigV4AuthSmithyClient : public AwsSmithyClientT<ServiceNameT, ServiceClientConfigurationT, smithy::SigV4AuthSchemeResolver<>, Aws::Crt::Variant<smithy::SigV4AuthScheme>, EndpointProviderT, smithy::client::JsonOutcomeSerializer, smithy::client::JsonOutcome>
```

to reduce the amount of generated boilerplate template arguments for the majority of services.

Or keep it as is.

Reply:
let me play around with it in the code generation revision, im tempted to keep it as is because i dont want `XmlClient` or `JsonClient` to exist as they had in the past but this would likely make sense from a boiler plate standpoint and since its just a alias for something else it makes sense.

Reply:
so spent some time messing around with this and im not opposed to it however see my [other comment](https://github.com/aws/aws-sdk-cpp/pull/3091/files#r1752638598). if we have to codegen either `AuthSchemeResolver` or the variant of `AuthSchemes` then we would have to make a unique class for each combination. yes sigv4 is the most common and captures most if not all of our use cases, but if more are added we will have to continue creating more and more exotic combinations of AuthSchemeResolver` and `AuthSchemes`.